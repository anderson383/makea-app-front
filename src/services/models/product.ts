import { ICategory } from "./category";

export interface IProduct {
  id: string;
  name: string;
  image: string;
  price: number;
  description: string;
  isAddCart?: boolean;
  categoria?: string | ICategory;
}