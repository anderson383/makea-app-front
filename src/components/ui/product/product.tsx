import { View, Text, StyleSheet, FlatList, ScrollView, Image, Button, TouchableNativeFeedback, TouchableOpacity, Touchable } from "react-native"
import { THEME } from "./../../../theme";
import { SvgXml } from 'react-native-svg';
import { IProduct } from "../../../services/models/product";
import { numberFormat } from "../../../helpers/numberCurrency";
import {HeartIcon, HeartIconActive} from "../../../../assets/icons/heart";

export interface ProductProps {
  product: IProduct
  onClickHeart?: () => void;
  onClickCard?: () => void;
}

const Product:React.FC<ProductProps> = ({product, onClickHeart, onClickCard}) => {
  return (
      <View style={styles.product} key={product.id}>
        <TouchableOpacity onPress={onClickCard && onClickCard} testID="productCard">
          <View style={{flex: 1}}>
            <Image source={{ uri: product.image, height: 150 }} style={{width: '100%'}} />
          </View>
          <View style={styles.info} >
            <View style={styles.priceContent}>
              <Text style={styles.price}>{numberFormat(product.price)}</Text>
              <TouchableOpacity  onPress={onClickHeart && onClickHeart} testID="addCart">
                { product.isAddCart ?  <SvgXml  xml={HeartIconActive()}  /> : <SvgXml  xml={HeartIcon()}  />} 
              </TouchableOpacity>
            </View>
            <View>
            <Text>{product.name}</Text>
            </View>
          </View>
        </TouchableOpacity>
      </View>
   
  )
}

const styles = StyleSheet.create({
  product: {
    marginRight: 3,
    borderRadius: 10,
    flexBasis: '47.8%',
    overflow: 'hidden',
    borderColor: '#f1f1f1',
    borderWidth: 1
  },
  info: {
    padding: 8,
    position: 'relative',
    gap: THEME.paddings.sm
  },
  priceContent: {
    flexDirection: 'row',
    justifyContent: 'space-between'
  },
  price: {
    color: THEME.colors.secondaryColor,
    fontWeight: 'bold'
  }
})

export default Product;