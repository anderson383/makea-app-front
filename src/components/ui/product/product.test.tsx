
import Product from '.'
import { act } from 'react-dom/test-utils';
import {render, screen, fireEvent, waitFor} from '@testing-library/react-native'
import { ProductProps } from './product';

const productMock = {
  id: 'id-test',
  image: 'imgaentest',
  description: 'product description',
  price: 2000,
  categoria: 'category',
  isAddCart: false,
  name: 'Chicharron! xd'
}

describe('Test shared Product Component', () => {
  
  const setup = async (props:ProductProps) => {
    return await waitFor(() => render(
      <Product {...props} />
    )) 
  }

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test(`Should render component`, async () => {
    const {toJSON} = await setup({
      product: productMock
    })

    expect(toJSON).toMatchSnapshot()
  })

  test(`Should render component is not add cart and click all card`, async () => {
    const pressAddCart = jest.fn()
    const clickCart = jest.fn()
    const {getByTestId} = await setup({
      product: {...productMock, isAddCart: true},
      onClickHeart: pressAddCart,
      onClickCard: clickCart
    })

    const butt = getByTestId('addCart')
    const card = getByTestId('productCard')
    
    fireEvent.press(butt)
    fireEvent.press(card)

    expect(pressAddCart).toHaveBeenCalled()
    expect(clickCart) .toHaveBeenCalled()
  })   
})
