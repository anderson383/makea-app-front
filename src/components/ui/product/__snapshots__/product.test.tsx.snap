// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test shared Product Component Should render component 1`] = `[Function]`;
